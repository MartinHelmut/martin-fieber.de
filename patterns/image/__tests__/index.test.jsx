import React from "react";
import renderer from "react-test-renderer";

import Image from "../index";

describe("Image", () => {
  test("should render", () => {
    const tree = renderer
      .create(
        <Image
          src="image.png"
          alt="Alternative text"
          width={500}
          height={500}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render with caption", () => {
    const tree = renderer
      .create(
        <Image
          src="image.png"
          alt="Alternative text"
          width={500}
          height={500}
          caption="An image caption"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
