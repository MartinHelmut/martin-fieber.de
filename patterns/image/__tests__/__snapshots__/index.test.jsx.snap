// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render 1`] = `
<figure
  className="image__Figure-lrfufk-0 jCBCVM"
>
  <img
    alt="Alternative text"
    className="image__Media-lrfufk-1 feNlVn"
    height={500}
    src="image.png"
    width={500}
  />
</figure>
`;

exports[`Image should render with caption 1`] = `
<figure
  className="image__Figure-lrfufk-0 jCBCVM"
>
  <img
    alt="Alternative text"
    className="image__Media-lrfufk-1 feNlVn"
    height={500}
    src="image.png"
    width={500}
  />
  <figcaption
    className="image__Caption-lrfufk-2 kDzRYO"
  >
    An image caption
  </figcaption>
</figure>
`;
